plugins/c8y_remote_access_plugin/src/main.rs

```diff
@@ -174,4 +176,30 @@ mod tests {
             Utf8PathBuf::from("/etc/tedge/operations/c8y/c8y_RemoteAccessConnect")
         );
     }
+
+    #[test]
+    fn cleanup_existing_operation() {
+        let dir = tempfile::tempdir().unwrap();
+
+        let operation_path = create_example_operation(dir.path().try_into().unwrap());
+        remove_supported_operation(dir.path().try_into().unwrap());
+
+        assert!(!operation_path.exists());
+    }
+
+    #[test]
+    fn cleanup_non_existent_operation() {
+        // Verify that this doesn't panic
+        remove_supported_operation(
+            "/tmp/already-deleted-operations-via-removal-of-tedge-agent".into(),
+        );
+    }
+
+    fn create_example_operation(dir: &Utf8Path) -> Utf8PathBuf {
+        let operation_path = supported_operation_path(dir);
+        std::fs::create_dir_all(operation_path.parent().unwrap()).unwrap();
+        std::fs::File::create(&operation_path).unwrap();
```

Comment:
Just for future reference: we've got our own wrapper over `tempfile` called `TempTedgeDir` which makes creation of hierarchical files slightly easier  as follows:

```
TempTedgeDir::new()
            .dir("operations")
            .dir("c8y")
            .file("c8y_RemoteAccessConnect")
            .with_raw_content("content")
```